developers/contributor-guide/weaviate-core/tests.md

```diff
@@ -87,29 +87,29 @@ go test ./...
 #### Adding new unit tests
 
 * Add unit tests in the same folder as the code they are testing
-* Aim to write "black box" unit tests which test the public ("exported") methods of the class under test without knowing too much about the internals
+* Aim to write "black box" unit tests that test the public ("exported") methods of the class under test without knowing too much about the internals.
 * Do not use any build tags.
 
 ### Integration tests
-As outline in the Philosophy, integration tests require backing services to be run. We have a convenience script available which starts all required services in `Docker Compose` and runs the tests against it:
+As outline in the Philosophy, integration tests require backing services to be run. We have a convenience script available which starts all required services in `docker compose` and runs the tests against it:
 
 ```sh
 test/integration/run.sh
 ```
 
 #### Adding new integration tests
 
-* Use the `integrationTest` build tag on your test, so it is ignored during unit test runs
-* Make sure the test prepares for and cleans up after itself, so tests can be run succession
-* If your test requires a lot of time to execute, consider marking them as a slow test and making them optional. (see below)
+* Use the `integrationTest` build tag on your test, so it is ignored during unit test runs.
+* Make sure the test prepares for and cleans up after itself, so tests can be run in succession.
+* If your test requires a lot of time to execute, consider marking them as a slow test and making them optional. (see below).
 
 #### Slow integration tests
 
 With the introduction of Standalone mode there is some behavior that needs to be tested at scale. For example, the HNSW index might work fine on a fictional test set where all entries are on layer 0, but then break once several layers need to be traversed.
 
 Similarly tests which test recall (in an HNSW index) require a dataset size, where the number of nodes is considerably larger than the `ef` parameter at search. Otherwise the search is a full-dataset scan which will always have 100% recall.
 
-These tests are considered "slow tests". They are an important part in our test pipeline, we wouldn't want to release Weaviate without them passing. However, they might not play a major role during every feature we develop. Therefore these test are opt-in with the `--include-slow` flag on the test runner:
+These tests are considered "slow tests". They are an important part in our test pipeline, we wouldn't want to release Weaviate without passing them. However, they might not play a major role during every feature we develop. Therefore these test are opt-in with the `--include-slow` flag on the test runner:
```

Comment:
Too avoid a possible misreading, "passing [over] them", consider:

They are an important part in our test pipeline. The slow tests are important for release testing, however, individual tests might not be important for feature testing. Therefore these tests ...